### ISSUE TITLE
Nodes launched by the cluster default node group or `NodeGroup` components do not register with cluster

### ISSUE THREAD

**Issue by flostadler** [pulumi/pulumi-eks#1356]
### What happened?

@desteves is running into issues with the pulumi template `kubernetes-aws-typescript`. The nodes launched by the default node group (using `NodeGroup` aka NodeGroupv1) do not seem to register with the cluster.  

### Example

https://github.com/pulumi/templates/blob/d1926471033be33b2f9c0266f23864854d74ac4c/kubernetes-aws-typescript/index.ts

### Output of `pulumi about`

```
CLI
Version      3.128.0
Go Version   go1.22.5
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  aws         6.51.0
resource  awsx        2.14.0
resource  docker      4.5.5
resource  docker      3.6.1
resource  eks         2.7.9
resource  kubernetes  4.17.1
language  nodejs      unknown

Host
OS       darwin
Version  14.5
Arch     arm64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by flostadler** [comment]
This is caused by https://github.com/pulumi/pulumi-aws/issues/4446. This affects AWS provider version `v6.51.0`.

You can fix this by downgrading to `v6.50.1`.
Note, only node is affected because it isn't using the MLC but directly interacts with the provider.

**Comment by desteves** [comment]
Your suggestion worked. Thank you!

Here's what I did:

* Add `"@pulumi/aws": "6.50.1"` to the `package.json` dependencies
* Run `npm install` (outputs "added 1 package, changed 1 package,...")
* Run ` pulumi up --yes --skip-preview` and confirm the output contains something like:
```bash
    Type                       Name                         Status        Info
     pulumi:pulumi:Stack       simple-k8s-cluster-dev                     2 messages
     ├─ eks:index:Cluster      eks-cluster
     │  ├─ aws:eks:Cluster     eks-cluster-eksCluster                     [diff: ~provider]

```

**Comment by flostadler** [comment]
This bug is fixed now in pulumi-aws [v6.51.1](https://github.com/pulumi/pulumi-aws/releases/tag/v6.51.1).
